assume/common/forecasts.py

```diff
@@ -178,6 +195,24 @@ def set_forecast(self, data: pd.DataFrame | pd.Series | None, prefix=""):
         else:
             self.forecasts[prefix + data.name] = data
 
+    def set_operation_states(self, unit: str, data: pd.DataFrame):
+        """
+        Sets the operational state data for a given unit.
+
+        Args:
+            unit (str): The unit for which the operational state data is provided.
+            data (pd.DataFrame): The operational state data.
+        """
+        # Check if the unit already exists in the operation_states dictionary
+        if unit in self.operation_states:
```

Comment:
```suggestion
        if unit in self.operation_states.keys():
```
I would prefer to be explicit here